docs/src/examples/neural_ode.md

```diff
@@ -40,15 +40,16 @@ end
 function loss_neuralode(p)
     pred = predict_neuralode(p)
     loss = sum(abs2, ode_data .- pred)
-    return loss, pred
+    return loss
 end
 
 # Do not plot by default for the documentation
 # Users should change doplot=true to see the plots callbacks
-callback = function (p, l, pred; doplot = false)
+function callback(state, l; doplot = false)
     println(l)
     # plot current prediction against data
     if doplot
+        pred = predict_neuralode(state.u)
```

Comment:
This is not a viable solution, it doubles the cost. We should show a better standard way to do this.

Reply:
It was the same before the gradient and loss were evaluated separately so I figured it's a fair tradeoff, the current alternative is to use a global variable.

Down the line I plan to improve OptimizationState so it can be hacked for these things, I am not sure how it should be yet but should be possible

Reply:
We should show a current best practice that does not involve recomputing since that can be quite a regression.